integration_tests/common.rs

```diff
@@ -104,9 +120,35 @@ macro_rules! merchant_cli {
 }
 pub(crate) use merchant_cli;
 
+pub fn store_db_state(party: &MaliciousParty, tag: &str) {
```

Comment:
Thank you for taking the generic approach here with a variable `Party`! I don't think we have a scenario where the merchant would roll back their database, though. The merchant database only changes when a channel is established or closed (e.g. not with each payment) so any rollbacks would only change indications of on-chain behaviors, which we assume are enforced by the contract. (e.g. trying to close again on a closed contract would just fail).

We can think more carefully about this - maybe there is a test that is best served by this kind of merchant rollback instead of a smart-contract specific test, but I think you can take this out and make this a customer-specific functionality.

Reply:
Ah that makes sense, I'll go ahead and remove the parameter and enum. Also in general, I made the new enum because there was already a `Party` enum in common.rs but it was used for the watchers and operations and didn't have a string representation that matched the file names.

Reply:
That makes sense, thank you for clarifying.

I realized now that the merchant database also holds the revocation secrets and nonces. So another behavior we could cause by rolling back is to make them "forget" about a payment that was already made. But this still doesn't lead to an attack (e.g. the merchant stealing money), I think, because a merchant close always has them just claim all the funds and isn't tied to a secret or nonce anyway.